qadence/constructors/rydberg_feature_maps.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+from sympy import Basic, Function
+
+from qadence import (
+    AbstractBlock,
+    AnalogRot,
+    AnalogRX,
+    AnalogRY,
+    AnalogRZ,
+    FeatureParameter,
+    Parameter,
+    kron,
+)
+from qadence.blocks import AnalogBlock
+from qadence.constructors.feature_maps import fm_parameter
+from qadence.logger import get_logger
+from qadence.types import BasisSet, TParameter
+
+logger = get_logger(__file__)
+
+AnalogRotationTypes = [AnalogRX, AnalogRY, AnalogRZ]
+
+
+def rydberg_tower_feature_map(
+    n_qubits: int,
+    param: str = "phi",
+    max_abs_detuning: float = 2 * np.pi * 10,
+    tower_weights: list[float] | None = None,
+) -> AbstractBlock:
+    """Feature map using semi-local addressing patterns.
+
+    Args:
+        n_qubits (int): number of qubits
+        param: the name of the feature parameter
+        max_abs_detuning: maximum value of absolute detuning for each qubit
+        tower_weights: a list of wegiths to assign to each qubit in the tower feature map
+
+    Returns:
+        AbstractBlock: _description_
+    """
+    max_abs_detuning = 2 * np.pi * 10
+    tower_coeffs = list(np.arange(1, n_qubits + 1)) if tower_weights is None else tower_weights
+    tower_detuning = max_abs_detuning / (sum(tower_coeffs[i] for i in range(n_qubits)))
+
+    param = FeatureParameter(param)
+    duration = 1000 * param / tower_detuning
+    return kron(
+        AnalogRot(
+            duration=duration,
+            delta=-tower_detuning * tower_coeffs[i],
+            phase=0.0,
+            qubit_support=(i,),
+        )
+        for i in range(n_qubits)
+    )
```

Comment:
Note that currently on main this will implement all the pulses for these rotations in sequence for pulser and in parallel for PyQ. Once #176 is merged both backends do it in sequence.

I guess this could be written with the `AddressingPattern` but it would fix the pattern for the whole circuit, not just the feature map.